Better Multithreading

SeeSoftware opened the pull request:
The code styling got a bit fucked because of my settings.

Commit:
static fix

hube12 committed:
Revert balint synchronous to async

hube12 committed:
Basic multithreading

Commit:
Added multithreading

Commit:
Reverted last commit

Commit:
Removed multi-threading

Commit:
Removed multi-threading

Commit:
Removed multi-threading

Commit:
Better speed display

SeeSoftware committed:
Multithreading

SeeSoftware committed:
Multithreading

Earthcomputer reviewed (changes requested):
Please undo the formatting changes

SeeSoftware committed:
Update main.cu

stylefix 1

SeeSoftware committed:
Update main.cu

stylefix 1

SeeSoftware committed:
Update main.cu

stylefix 2

SeeSoftware committed:
Update main.cu

stylefix 2

SeeSoftware committed:
Update main.cu

stylefix 3

SeeSoftware committed:
Update main.cu

stylefix 3

SeeSoftware committed:
Update main.cu

stylefix 4

SeeSoftware committed:
Update main.cu

stylefix 4

SeeSoftware commented:
styling should be corrected now.

SeeSoftware committed:
cleaning

SeeSoftware committed:
cleaning

SeeSoftware committed:
fix msvc defines

SeeSoftware committed:
fix msvc defines

SeeSoftware committed:
Update main.cu

compile error fixes

SeeSoftware committed:
Update main.cu

compile error fixes

hube12 committed:
Fix memory leak by balint

human-0 committed:
Removed unnecessary heap allocation

hube12 committed:
Window fix

hube12 committed:
Fix looping counter (bad size)

BalintCsala committed:
Merge pull request #2 from human-0/master

Removed unnecessary heap allocation

Commit:
Removed unneccessary delete[]

hube12 committed:
Make nvcc shut up about printf wrong sizes

riking committed:
Unroll generateLeafPattern and only write 0, 4

Unfortunately this may or may not be faster due to icache, testing is required to verify.

riking committed:
Remove pow2 check for random calls that are pow2

riking committed:
Write all leaf results but keep unrolled

Minimal performance hit.

riking committed:
Merge remote-tracking branch 'SeeSoftware/multithreaded' into mt-opt

riking commented on main.cu:
```diff
@@ -401,62 +389,67 @@ int main(int argc, char *argv[]) {
+        static auto threadFunc = [&](size_t start, size_t end) {
+            for (int j = start; j < end; ++j) {
+                if (generator::ChunkGenerator::populate(tempStorage[j], X_TRANSLATE + 16)) {
+                    std::lock_guard<std::mutex> lock(fileMutex);
```
This looks like it's going to have a heck of a lot of mutex contention. Could the threads batch up their file writes?

BalintCsala committed:
Merge pull request #4 from riking/patch-2

Optimizations

riking committed:
Improve file write buffering

riking committed:
Improve file write buffering

riking committed:
Use wall clock time for progress reports

Also, use the PRIu64 macro to fix the printf format debacle once and for all

riking committed:
Merge branch 'wall-clock' into mt-opt

Conflicts:
	main.cu

riking committed:
fix compile for atomics

riking committed:
fix compile for atomics

riking committed:
Merge branch 'mt-opt'

riking committed:
fixup to write buffer optimizations

riking committed:
fixup to write buffer optimizations

human-0 committed:
Added uncheched mask for LCG modulo

human-0 committed:
Removed extra calculations for ignored leaf results

human-0 committed:
Removed extra calculations for ignored leaf results

human-0 committed:
Removed unnecessary leaf pattern generation

human-0 committed:
Used lcg.combine to optimise skipping

hube12 committed:
Merge pull request #6 from riking/wall-clock

[Major] Use wall clock time for progress reports

hube12 committed:
Removed unecessary d

Earthcomputer committed:
Create README.md

Commit:
Merge branch 'master' of https://github.com/human-0/seed-reversal-merged into human-0-master

Commit:
Merge branch 'human-0-master'

Commit:
Merged #7

Commit:
Merge branch 'riking-mt-opt-pr'

BalintCsala committed:
Revert "Multithreading + file write optimizations"

BalintCsala committed:
Merge pull request #11 from pack-png-mods/revert-7-mt-opt-pr

Revert "Multithreading + file write optimizations"

Commit:
Standardized

SeeSoftware committed:
lockfree thread loop

SeeSoftware committed:
Merge remote-tracking branch 'upstream/master' into multithreaded

SeeSoftware committed:
lockfree thread loop

about 7% performance boost on my maschine.
